src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -1475,1524 +1574,1682 @@ int16_t LoRaWANNode::downlink(uint8_t* data, size_t* len, LoRaWANEvent_t* event)
+      }
+    }
+  } else {    // bandType == RADIOLIB_LORAWAN_BAND_FIXED
+    LoRaWANChannel_t chnl = RADIOLIB_LORAWAN_CHANNEL_NONE;
+    uint8_t spanNum = 0;
+    int chNum = 0;
+    int chOfs = 0;
+    for(; chNum < 64; chNum++) {
+      if(chMaskGrp0123 & ((uint64_t)1 << chNum)) {
         chnl.enabled = true;
         chnl.idx   = chNum;
-        chnl.freq  = this->band->txSpans[1].freqStart + i*this->band->txSpans[1].freqStep;
-        chnl.drMin = this->band->txSpans[1].drMin;
-        chnl.drMax = this->band->txSpans[1].drMax;
-        this->availableChannels[RADIOLIB_LORAWAN_CHANNEL_DIR_UPLINK][idx++] = chnl;
+        chnl.freq  = this->band->txSpans[spanNum].freqStart + chNum*this->band->txSpans[spanNum].freqStep;
+        chnl.drMin = this->band->txSpans[spanNum].drMin;
+        chnl.drMax = this->band->txSpans[spanNum].drMax;
+        this->channelPlan[RADIOLIB_LORAWAN_UPLINK][num++] = chnl;
       }
     }
-
-  }
-  if(this->band->numTxSpans == 2 && chMaskCntl == 7) {
-    // all channels off (clear all channels)
-    chnl = RADIOLIB_LORAWAN_CHANNEL_NONE;
-    for(int i = 0; i < RADIOLIB_LORAWAN_NUM_AVAILABLE_CHANNELS; i++) {
-      this->availableChannels[RADIOLIB_LORAWAN_CHANNEL_DIR_UPLINK][i] = chnl;
-      // downlink channels are not defined so don't need to reset
-    }
-    idx = 0;
-    // a '1' enables a single channel from second span
-    for(uint8_t i = 0; i < 8; i++) {
-      uint16_t mask = 1 << i;
-      if(mask & chMask) {
-        // enable single channel from second span
-        uint8_t chNum = 64 + i;
+    if(this->band->numTxSpans > 1) {
+      spanNum += 1;
+      chNum = 0;
+      chOfs = 64;
+    }
+    for(; chNum < this->band->txSpans[spanNum].numChannels; chNum++) {
+      if(chMaskGrp45 & ((uint32_t)1 << chNum)) {
         chnl.enabled = true;
-        chnl.idx   = chNum;
-        chnl.freq  = this->band->txSpans[1].freqStart + i*this->band->txSpans[1].freqStep;
-        chnl.drMin = this->band->txSpans[1].drMin;
-        chnl.drMax = this->band->txSpans[1].drMax;
-        this->availableChannels[RADIOLIB_LORAWAN_CHANNEL_DIR_UPLINK][idx++] = chnl;
+        chnl.idx   = chNum + chOfs;
+        chnl.freq  = this->band->txSpans[spanNum].freqStart + chNum*this->band->txSpans[spanNum].freqStep;
+        chnl.drMin = this->band->txSpans[spanNum].drMin;
+        chnl.drMax = this->band->txSpans[spanNum].drMax;
+        this->channelPlan[RADIOLIB_LORAWAN_UPLINK][num++] = chnl;
       }
     }
-
   }
 
-  #if RADIOLIB_DEBUG_PROTOCOL
+#if RADIOLIB_DEBUG_PROTOCOL
   this->printChannels();
-  #endif
+#endif
 
   return(true);
 }
 
-uint8_t LoRaWANNode::getMacPayloadLength(uint8_t cid) {
-  for (LoRaWANMacSpec_t entry : MacTable) {
-    // cppcheck warns here we should use std::find_if, but some platforms may not have that
-    if (entry.cid == cid) { // cppcheck-suppress useStlAlgorithm
-      return entry.lenDn;
+void LoRaWANNode::printChannels() {
```

Comment:
Because this only actually prints in debug mode, I think it would be better to guard this with `#if RADIOLIB_DEBUG_PROTOCOL`.